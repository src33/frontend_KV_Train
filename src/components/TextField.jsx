// import { useEffect, useState } from "react";

import React from "react";
import { forwardRef, useState } from "react";
const TextField = forwardRef((props, ref) => {
   // const [disabled, setDisabled] = useState(false);
   const [defVal, setDefVal] = useState(props.value);
   
   const onChange = (e) => {
      setDefVal(e.target.value);
      if (props.onChange) props.onChange(e.target.value, props.label);
   };
   // console.log(props.disabled)

   return (
      <span className={props.className}>
         <label htmlFor={props.label} data-testid="textFieldLabel-test-id">{props.label}</label>
         <input
            type={props.type}
            ref={ref}
            name={props.label}
            value={defVal || props.value}
            placeholder={props.label}
            onChange={onChange}
            disabled={props.disabled?true:false}
            data-testid="textField-test-id"
         />
         <label htmlFor="error" className="error_msg">
            {props.error}
         </label>
      </span>
   );
});
export default TextField;
