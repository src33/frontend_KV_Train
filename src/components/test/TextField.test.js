import React from "react";
import { render, fireEvent, getByLabelText } from "@testing-library/react";
import TextField from "../TextField";

describe("Check if textfields are working properly", () => {
   test("Check if textfields are rendered", () => {
      const { getByTestId } = render(<TextField />);
      const element = getByTestId("textField-test-id");

      expect(element).toBeTruthy();
   });
   test("Check if value is displayed correctly", () => {
      const text = "value";
      const { getByDisplayValue } = render(<TextField value={text} />);
      getByDisplayValue(text);
   });
   test("Check if placeholder is displayed correctly", () => {
      const text = "placeholder";
      const { getByPlaceholderText } = render(<TextField label={text} />);
      getByPlaceholderText(text);
   });
   test("Check if onChange is triggered", () => {
      const onChange = jest.fn();
      const { getByTestId } = render(<TextField onChange={onChange} />);
      const element = getByTestId("textField-test-id");
      fireEvent.change(element, { target: { value: "a" } });

      expect(onChange).toHaveBeenCalled();
   });
   test("Check input is received", () => {
      const { getByTestId } = render(<TextField value="" />);
      const element = getByTestId("textField-test-id");

      fireEvent.change(element, { target: { value: "sample" } });
      expect(element.value).toBe("sample");
   });
   test.skip("Check if labels are displayed correctly", () => {
      const text = "label";
      const { } = render(<TextField label={text} />);
      const element =  getByTestId("textFieldLabel-test-id");
        const {getByLabelText}=
      expect(getByLabelText()).toBe(text);
   });
   test("Check if snapshots is matched properly", () => {
    const onChange = jest.fn();
    const text = "value";
    const { asFragment } = render(<TextField onChange={onChange} value={text}/>);
    
    expect(asFragment()).toMatchSnapshot();
 });

});
