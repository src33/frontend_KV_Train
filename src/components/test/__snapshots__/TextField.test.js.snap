// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if textfields are working properly Check if snapshots is matched properly 1`] = `
<DocumentFragment>
  <span>
    <label
      data-testid="textFieldLabel-test-id"
    />
    <input
      data-testid="textField-test-id"
      value="value"
    />
    <label
      class="error_msg"
      for="error"
    />
  </span>
</DocumentFragment>
`;
